hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/OperationServiceImpl.java

```diff
@@ -460,9 +460,23 @@ public void provideMetrics(MetricsRegistry registry) {
     public void start() {
         logger.finest("Starting OperationService");
 
-        ManagedExecutorService asyncExecutor = nodeEngine.getExecutionService().register(
-                ExecutionService.ASYNC_EXECUTOR, Runtime.getRuntime().availableProcessors(),
-                ASYNC_QUEUE_CAPACITY, ExecutorType.CONCRETE);
+        setInvocationContext();
+
+        invocationMonitor.start();
+        operationExecutor.start();
+        asyncResponseHandler.start();
+        slowOperationDetector.start();
+    }
+
+    public void setInvocationContext() {
+        ManagedExecutorService asyncExecutor;
+        if (this.invocationContext != null) {
```

Comment:
Is initInvocationContext called more than once? Because there is a null check I don't understand.

If it is called more than once, there could be 2 instances of the invocationContext which doesn't make a lot of sense.